src/k8s/pkg/k8sd/setup/kube_apiserver.go

```diff
@@ -86,13 +87,17 @@ func KubeAPIServer(snap snap.Snap, serviceCIDR string, authWebhookURL string, en
 		"--tls-private-key-file":                     path.Join(snap.KubernetesPKIDir(), "apiserver.key"),
 	}
 
+	if nodeIP != nil && !nodeIP.IsLoopback() {
+		args["--advertise-address"] = nodeIP.String()
+	}
+
 	switch datastore.GetType() {
 	case "k8s-dqlite", "external", "etcd":
 	default:
 		return fmt.Errorf("unsupported datastore %s, must be one of %v", datastore.GetType(), SupportedDatastores)
 	}
 
-	datastoreUpdateArgs, deleteArgs := datastore.ToKubeAPIServerArguments(snap, []string{nodeIP})
+	datastoreUpdateArgs, deleteArgs := datastore.ToKubeAPIServerArguments(snap, []string{nodeIP.String()})
```

Comment:
@neoaggelos I think we are reverting this change anyways, but are you happy with getting the nodeIP from nodeIP passed in as `net.IP` so we can share it?